.github/workflows/build.yml

```diff
@@ -21,20 +21,11 @@ jobs:
 
       - name: Install Rust
         uses: dtolnay/rust-toolchain@stable
-        with:
-          toolchain: nightly # nightly is required for fmt
-          components: rustfmt, clippy
-
-      - name: Check
-        run: cargo check
 
-      - name: Format
-        run: |
-          cargo fmt --all -- --check
-          cargo clippy --workspace --bins -- -D warnings
-
-      - name: Build Debug
-        run: cargo build --verbose
```

Comment:
@Pushkarm029 @kirillt I removed those steps for now, because the behavior of the `Format` was not being 100% predictable. Since I started this PR, around a month ago, the `Format` step changed its results multiple times, even though the Rust code remained exactly the same

As example, you can check that from time to time some old and recent jobs didn't fully complete because formatting errors were randomly being thrown.

Most recent occurrence:
- https://github.com/ARK-Builders/ark-core/actions/runs/12120246957/job/33788525665 ran just fine
- https://github.com/ARK-Builders/ark-core/actions/runs/12287131386/job/34288596396 a week later started raising a formatting error

I judge as a bad practice using a `nightly` or any other "dynamic" version to format a project, since it can introduce errors from newly added formatting features :thinking: 

Curious to know what you guys think about it

Reply:
Interesting point. Actually, I prefer setting up `cargo fmt` everywhere, because it helps to avoid inconsistent code style (and standard Rust code style looks pretty fine). But I've never encountered `cargo fmt check` rejecting a bit outdated codebase.

**My guess** is that the code style used by `cargo fmt` develops including new decisions made by Rust community. And out code is modified too slowly so we don't include these new updates. If this is true, then we could keep `cargo fmt check` and actually reformat the code every time we encounter such failures on CI, but only manually.

@Pushkarm029 what do you think?

Reply:
We need to use nightly for formatting because some of the rules we define in `rustfmt.toml` are only supported in the nightly version. Rust nightly is updated frequently, that’s not a problem — it’s the pre-release version. We have two options:
	•	We can abandon the rules we’ve defined (I would like to keep them)
	•	We can check for formatting in a separate job that runs nightly, within the same workflow.
For example:
```yaml
jobs:
  format:
    name: Code Formatting Check
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4

      - name: Install Nightly Rust with rustfmt
        uses: dtolnay/rust-toolchain@stable
        with:
          toolchain: nightly
          components: rustfmt

      - name: Run rustfmt
        run: cargo fmt --all -- --check

  build_and_test:
    name: Build and Test
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4

      - name: Install Stable Rust
        uses: dtolnay/rust-toolchain@stable
        with:
          toolchain: stable

      ...
```

Reply:
We actually care about formatting only during merging, so we probably can configure GitHub rules and prevent merge if `cargo fmt` job fails specifically. This would allow see that the build itself works.